intel-lustre-clients-vmss-centos/azuredeploy.json

```diff
@@ -0,0 +1,414 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "artifactsBaseUrl": {
+      "type": "string",
+      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/intel-lustre-clients-vmss-centos",
+      "metadata": {
+        "description": "Base URL of the solution template gallery package",
+        "artifactsBaseUrl": ""
+      }
+    },
+    "location": {
+      "type": "string",
+      "allowedValues": [
+        "West US",
+        "East US 2",
+        "East US",
+        "Central US",
+        "South Central US",
+        "North Central US",
+        "West Europe",
+        "North Europe",
+        "East Asia",
+        "Southeast Asia",
+        "Japan West",
+        "Japan East",
+        "Brazil South",
+        "Australia East",
+        "Australia Southeast",
+        "Central India",
+        "West India",
+        "South India"
+      ],
+      "metadata": {
+        "description": "Location for the Lustre nodes"
+      }
+    },
+    "imageSku": {
+      "type": "string",
+      "defaultValue": "6.6",
+      "allowedValues": [
+        "6.6",
+        "7.0"
+      ],
+      "metadata": {
+        "description": "OpenLogic CentOS version to use"
+      }
+    },
+    "vmssName": {
+      "type": "string",
+      "metadata": {
+        "description": "String used as a base for naming resources (9 characters or less). A hash is prepended to this string for some resources, and resource-specific information is appended."
+      },
+      "maxLength": 9
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Admin username for the virtual machines"
+      }
+    },
+    "authenticationType": {
+      "type": "string",
+      "defaultValue": "password",
+      "allowedValues": [
+        "password",
+        "sshPublicKey"
+      ],
+      "metadata": {
+        "description": "Authentication type for the virtual machines"
+      }
+    },
+    "adminPassword": {
```

Comment:
For this specific template, to be able to provide "either-or" functionality, so either SSH -or- the password can be used, both parameters have to support empty default values. The user will provide the proper value for the parameter based on what they select for authenticationType (since there is no way to have a conditional required flag on parameters)